### PR TITLE
HW GlobalExceptionHandler and Data Validation

### PR DESCRIPTION
**Pull request by katrevichdv**

### CODE REVIEW

File: src/main/java/com/example/bookstore/exception/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<String> handleAllExceptions(Exception ex) {
+        return new ResponseEntity<>("An unexpected error occurred",
+                HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @ExceptionHandler(NullPointerException.class)
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/exception/GlobalExceptionHandler.java]: remove, you shouldn't get such exception

File: src/main/java/com/example/bookstore/exception/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        List<String> errors = ex.getBindingResult()
+                .getAllErrors().stream()
+                .map(DefaultMessageSourceResolvable::getDefaultMessage)
+                .collect(Collectors.toList());
+        return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<String> handleAllExceptions(Exception ex) {
+        return new ResponseEntity<>("An unexpected error occurred",
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/exception/GlobalExceptionHandler.java]: provide more details about exception

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
let's fix some comments

File: src/main/java/com/example/bookstore/dto/BookRequestDto.java [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
+import jakarta.validation.constraints.NotBlank;
+
 public record BookRequestDto(
+        @NotBlank
         String title,
+        @NotBlank
         String author,
+        @NotBlank
         String isbn,
+        @Min(0) @Digits(integer = 8, fraction = 2)
```
Comment (MentorMateAcademy) [inline src/main/java/com/example/bookstore/dto/BookRequestDto.java]:
```suggestion
        @Positive
```

File: src/main/java/com/example/bookstore/dto/BookRequestDto.java [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
         String price,
         String description,
```
Comment (MentorMateAcademy) [inline src/main/java/com/example/bookstore/dto/BookRequestDto.java:21]: You can validate description

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookstore/dto/BookRequestDto.java (hunk 4 above)
Comment (katrevichdv) [reply src/main/java/com/example/bookstore/dto/BookRequestDto.java:21]: There is only one constraint about max length of description field in db.
Adding NotBlank annotation would make unavailable to update prices of exsisting books with empty desciption. 
Here are a few options: 
Leave it as is.
Add Length annotation
Update changeSet, Model and DTO

Comment (Elena-Bruyako) [reply src/main/java/com/example/bookstore/dto/BookRequestDto.java:21]: You can just add `@Size `

File: src/main/java/com/example/bookstore/dto/BookSearchParametersDto.java [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
 
+import jakarta.validation.constraints.Digits;
+import jakarta.validation.constraints.Positive;
 import java.util.List;
 
 public record BookSearchParametersDto(
         List<String> title,
         List<String> author,
         List<String> isbn,
+        @Positive @Digits(integer = 8, fraction = 2)
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/dto/BookSearchParametersDto.java:12]: Positive annotaiton works only with numeric values

File: src/main/java/com/example/bookstore/exception/GlobalExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        errors.add(ex.getMessage());
+        Arrays.stream(ex.getSuppressed())
+                .map(Throwable::getMessage)
+                .forEach(errors::add);
+        return new ResponseEntity<>(errors, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    public ResponseEntity<String> handleEntityNotFoundException(EntityNotFoundException ex) {
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(ex.getMessage());
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/exception/GlobalExceptionHandler.java]:
```suggestion
        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(ex.getMessage());
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's fix some comments

File: src/main/java/com/example/bookstore/dto/BookSearchParametersDto.java (hunk 5 above)
Comment (katrevichdv) [reply src/main/java/com/example/bookstore/dto/BookSearchParametersDto.java:12]: You must be kidding me.

**Review by MentorMateAcademy — APPROVED** [review]

**Pull request merged**